testing/internal/e2e/boundary/session_recording.go

```diff
@@ -0,0 +1,116 @@
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var sessionRecordingListResult sessionrecordings.SessionRecordingListResult
+	err := json.Unmarshal(output.Stdout, &sessionRecordingListResult)
+	require.NoError(t, err)
+
+	return sessionRecordingListResult.Items[0].Id
+}
+
+// WaitForChannelRecordingCli uses boundary CLI to periodically read a session recording by ID
+// and check if it has at least one connection and channel.
+// The method will fail if there are 0 channels found or return channel ID otherwise.
+func WaitForChannelRecordingCli(t testing.TB, ctx context.Context, sessionId string) string {
+	t.Log("Waiting for channel recording to appear...")
+	var channel *sessionrecordings.ChannelRecording
+	err := backoff.RetryNotify(
+		func() error {
+			output := e2e.RunCommand(ctx, "boundary",
+				e2e.WithArgs(
+					"session-recordings", "read",
+					"-id", sessionId,
+					"-format", "json",
+				),
+			)
+			if output.Err != nil {
+				return backoff.Permanent(errors.New(string(output.Stderr)))
+			}
+
+			var sessionRecordingReadResult sessionrecordings.SessionRecordingReadResult
+			err := json.Unmarshal(output.Stdout, &sessionRecordingReadResult)
+			if err != nil {
+				return backoff.Permanent(err)
+			}
+
+			connectionCount := len(sessionRecordingReadResult.Item.ConnectionRecordings)
+			if connectionCount == 0 {
+				return errors.New("No items are appearing in the connection recording list")
+			}
+
+			channelCount := len(sessionRecordingReadResult.Item.ConnectionRecordings[0].ChannelRecordings)
+			if channelCount == 0 {
+				return errors.New("No items are appearing in the channel recording list")
+			}
+
+			channel = sessionRecordingReadResult.Item.ConnectionRecordings[0].ChannelRecordings[0]
+			return nil
+		},
+		backoff.WithMaxRetries(backoff.NewConstantBackOff(3*time.Second), 5),
+		func(err error, td time.Duration) {
+			t.Logf("%s. Retrying...", err.Error())
+		},
+	)
+	require.NoError(t, err)
+	t.Logf("Found Channel Recording: %s", channel.Id)
+	return channel.Id
+}
+
+// WaitAndDownloadSessionRecordingFile uses boundary CLI to periodically attempt to download
+// a session recording file from AWS S3 Bucket.
+// The method will fail if the file is not available on AWS yet or downloading failed.
+// It will return downloaded file name otherwise. The file will be saved in the current directory.
+func WaitAndDownloadSessionRecordingFile(t testing.TB, ctx context.Context, channelId string) string {
+	t.Log("Waiting for session recording file to download it...")
+	err := backoff.RetryNotify(
+		func() error {
+			output := e2e.RunCommand(ctx, "boundary",
+				e2e.WithArgs(
+					"session-recordings", "download",
+					"-id", channelId,
+					"-format", "json",
+				))
+			return output.Err
+		},
+		backoff.WithMaxRetries(backoff.NewConstantBackOff(3*time.Second), 5),
+		func(err error, td time.Duration) {
+			t.Logf("%s. Retrying...", err.Error())
+		},
+	)
+	require.NoError(t, err)
```

Comment:
Instead of retrying this until it passes, what do you think about retrying `read` on the session recording and checking the `state`? When it's `available` the download should work first time.